src/globus_sdk/services/gcs/connector_table.py

```diff
@@ -0,0 +1,147 @@
+
+    .. code-block:: pycon
+
+        >>> connector = ConnectorTable.AZURE_BLOB
+        >>> connector.name
+        'Azure Blob'
+        >>> connector.connector_id
+        '9436da0c-a444-11eb-af93-12704e0d6a4d'
+    """
+
+    POSIX: t.ClassVar[GlobusConnectServerConnector] = GlobusConnectServerConnector(
+        name="POSIX", connector_id="145812c8-decc-41f1-83cf-bb2a85a2a70b"
+    )
+    AZURE_BLOB: t.ClassVar[GlobusConnectServerConnector] = GlobusConnectServerConnector(
+        name="Azure Blob", connector_id="9436da0c-a444-11eb-af93-12704e0d6a4d"
+    )
+    BLACKPEARL: t.ClassVar[GlobusConnectServerConnector] = GlobusConnectServerConnector(
+        name="BlackPearl", connector_id="7e3f3f5e-350c-4717-891a-2f451c24b0d4"
+    )
+    BOX: t.ClassVar[GlobusConnectServerConnector] = GlobusConnectServerConnector(
+        name="Box", connector_id="7c100eae-40fe-11e9-95a3-9cb6d0d9fd63"
+    )
+    CEPH: t.ClassVar[GlobusConnectServerConnector] = GlobusConnectServerConnector(
+        name="Ceph", connector_id="1b6374b0-f6a4-4cf7-a26f-f262d9c6ca72"
+    )
+    DROPBOX: t.ClassVar[GlobusConnectServerConnector] = GlobusConnectServerConnector(
+        name="Dropbox", connector_id="49b00fd6-63f1-48ae-b27f-d8af4589f876"
+    )
+    ONEDRIVE: t.ClassVar[GlobusConnectServerConnector] = GlobusConnectServerConnector(
+        name="OneDrive", connector_id="28ef55da-1f97-11eb-bdfd-12704e0d6a4d"
+    )
+    GOOGLE_DRIVE: t.ClassVar[GlobusConnectServerConnector] = (
+        GlobusConnectServerConnector(
+            name="Google Drive", connector_id="976cf0cf-78c3-4aab-82d2-7c16adbcc281"
+        )
+    )
+    GOOGLE_CLOUD_STORAGE: t.ClassVar[GlobusConnectServerConnector] = (
+        GlobusConnectServerConnector(
+            name="Google Cloud Storage",
+            connector_id="56366b96-ac98-11e9-abac-9cb6d0d9fd63",
+        )
+    )
+    ACTIVESCALE: t.ClassVar[GlobusConnectServerConnector] = (
+        GlobusConnectServerConnector(
+            name="ActiveScale", connector_id="7251f6c8-93c9-11eb-95ba-12704e0d6a4d"
+        )
+    )
+    S3: t.ClassVar[GlobusConnectServerConnector] = GlobusConnectServerConnector(
+        name="S3", connector_id="7643e831-5f6c-4b47-a07f-8ee90f401d23"
+    )
+    POSIX_STAGING: t.ClassVar[GlobusConnectServerConnector] = (
+        GlobusConnectServerConnector(
+            name="POSIX Staging", connector_id="052be037-7dda-4d20-b163-3077314dc3e6"
+        )
+    )
+    IRODS: t.ClassVar[GlobusConnectServerConnector] = GlobusConnectServerConnector(
+        name="iRODS", connector_id="e47b6920-ff57-11ea-8aaa-000c297ab3c2"
+    )
+    HPSS: t.ClassVar[GlobusConnectServerConnector] = GlobusConnectServerConnector(
+        name="HPSS", connector_id="fb656a17-0f69-4e59-95ff-d0a62ca7bdf5"
+    )
+
+    @classmethod
+    def all_connectors(cls) -> t.Iterable[GlobusConnectServerConnector]:
+        """
+        Return an iterator of all known connectors.
+        """
+        for item in vars(cls).values():
```

Comment:
This feels like the sort of move I would make, so while I don't take any issue w/ this, I also support (per our conversation) getting @kurtmckee's input on whether this is potentially problematic or whether there is a better, more idiomatic alternative we should reach for.

Reply:
I think that I would reach for this implementation, since I like to keep raw data as raw data:

```python
class ConnectorTable:
    AZURE_BLOB: t.ClassVar[GlobusConnectServerConnector]
    ACTIVESCALE: t.ClassVar[GlobusConnectServerConnector]
    ...

    _connectors = (
        ("AZURE_BLOB", "Azure Blob", "9436da0c-a444-11eb-af93-12704e0d6a4d"),
        ("ACTIVESCALE", "Active Scale", "7251f6c8-93c9-11eb-95ba-12704e0d6a4d"),
        ...
    )

    for _attribute, _name, _id in _connectors:
        locals()[_attribute] = GlobusConnectServerConnector(name=_name, connector_id=_id)
    del _attribute, _name, _id

    @classmethod
    def all_connectors(cls) -> ...:
        yield from (getattr(cls, attribute) for attribute, _, _ in cls._connectors)
```

This accommodates name aliases and name changes, and means that the underlying data don't have to be duplicated/synchronized elsewhere in the test suite.

Reply:
I can do something like this -- I'll work on it now -- but I'll note that I'm not convinced that this is a net simplification.

I'm okay with the data being defined here as a tuple of tuples, but then I need to add some testing which validates that all of the declared classvars get assigned values and that all of the values have a corresponding classvar declaration.

If the real target here is the duplication in the testsuite, then I think this all makes sense to me. I'm not that bothered by having a replica of data in tests in order to validate a data container, but I'm open to eliminating it as long as we don't introduce new possibilities for discrepancies.

Reply:
https://github.com/globus/globus-sdk-python/pull/955/commits/cc83ef1cfcaffb7bde03f0bafb9781692282cb6f is my interpretation of this request.
I learned today about `inspect.get_annotations()`, which worked out to be much simpler for this kind of inspection than `typing.get_type_hints()`. So the main thing I was concerned about -- testsuite complexity -- is handled nicely.

Since this implementation is pretty close to what you requested, I'm marking this resolved. LMK if there are further tweaks on this front though.